### PR TITLE
Sort company search results by "importance"

### PR DESCRIPTION
**Pull request by jhigginbotham64**
@krit-gpt told me that @julianjear wanted the company search results to be sorted by a score, which was to be calculated as a weighted sum of the number of salaries, job ads, and reviews for a company. This is what I've done with the code below, I think it's simple enough to be readable, and it seems like it works. Let me know if the "algorithm" needs to be changed, or if you can spot any other problems.

There are a few peculiarities noted in the comments, please post a comment if further explanation is required.

Here are some helpful links:

Array sorting in JavaScript:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort

Deep-copying arrays in JavaScript:
https://stackoverflow.com/questions/597588/how-do-you-clone-an-array-of-objects-in-javascript

### CODE REVIEW

File: meteor-app/imports/ui/pages/company-search-trial.jsx [hunk 1]
```diff
@@ -50,23 +50,25 @@ const SearchResults = ({ searchText }) => (
...
+			// deep copy before we mutate it with sort:
+			// https://stackoverflow.com/questions/597588/how-do-you-clone-an-array-of-objects-in-javascript
+			const resultList = data.searchCompanies
+			.map(c => Object.assign({}, c))
+			// Array.prototype.sort is in-place and returns the new array
+			.sort(function(a, b) {
+				// This scoring method was given to me by Krit,
+				// who told me that Julian wanted it this way.
+				const score = (company) => (company.numJobAds * 2) + (company.numReviews * 1.5) + (company.numSalaries * 1);
+				const aScore = score(a);
```
Comment (julianjear) [inline meteor-app/imports/ui/pages/company-search-trial.jsx]: Only thing I think I'm not understanding is why there is an aScore and a bScore. What is the difference? I imagine that a and b are both two companies?

**Comment by julianjear** [comment]
Also, the algorithm values (+2 for jobs, +1.5 for reviews, ....) is not necessarily "what I want" but rather a suggestion. Please feel free to suggest different values for the algorithm if you think they are more reasonable, we can definitely have a discussion about it.
Great work!

File: meteor-app/imports/ui/pages/company-search-trial.jsx (hunk 1 above)
Comment (jhigginbotham64) [reply meteor-app/imports/ui/pages/company-search-trial.jsx]: Okay...
Notice that they occur as part of the argument to Array.prototype.sort. sort takes a sorting function as an argument. The docs explain what arguments the sorting function can take and what values it should return in order for the array to be sorted "properly". This allows you to sort arrays however you want.
So sort takes an argument (optionally) which is a function. That function which it takes as an argument, the "sorting function", is supposed to take two arguments: the items being compared. Now at any given point when the sorting function is called, its arguments could theoretically be any pair of items taken from the array being sorted. Lucky for you, in this case all the items in the array are companies.
So how are we specifically here at Vize wanting to determine which company comes first in the results list? Whichever one has the higher "score". The "scoring algorithm" was given to me by @krit-gpt , who says he got it from you.
Does that all make sense?

Comment (jhigginbotham64) [reply meteor-app/imports/ui/pages/company-search-trial.jsx]: There are actually links to some relevant documentation in the pull request description...

**Comment by jhigginbotham64** [comment]
Okay. Well, if I need to change the comments to better reflect the situation, please let me know. My comments and descriptions reflect the impressions I was under when I talked to Krit about it. I have no complaints about your suggested scoring method, it seems as reasonable as anything for the moment, so I was okay with pretty much "doing what I was told".
I'm definitely open to further discussion, but really I have no opinion about if/when this should be merged. It may be a good idea to let others weigh in, but...as the project/product manager, it's kind of your call. 
Though I will say that this particular feature will be trivially easy to fix or remove if we decide we don't like it for some reason.

File: meteor-app/imports/ui/pages/company-search-trial.jsx (hunk 1 above)
Comment (julianjear) [reply meteor-app/imports/ui/pages/company-search-trial.jsx]: Okay awesome, thanks for taking the time to explain that makes sense, I had a feeling the two scores would be used to be compared against each other. I'll take a look at some of the documentation as well to understand how the sort function works more in depth.

**Comment by julianjear** [comment]
No need to change the comments, I just don't want you guys to feel like I am imposing things upon you. I think it is more productive to suggest solutions and get input from the team to then come to a final decision instead of just doing what I believe is best. This is actually something I should have brought up at the meeting to get other people's suggestions but you're right, this can be changed later if needed so it's good the way it is now. I will wait until @krit-gpt reviews this pull request so he can take a look at it and accept + merge.

**Pull request merged**